concord-server/server/src/main/java/com/rabidgremlin/concord/resources/PhrasesResource.java

```diff
@@ -215,15 +215,14 @@ public Response voteForPhrase(@ApiParam(hidden = true) @Auth Caller caller, @Pat
 
   private void castVote(String phraseId, String label, String userId)
   {
-    int voteCountForPhrase = votesDao.getVoteCountForPhrase(phraseId);
-
     votesDao.upsert(phraseId, label, userId);
 
     // are we marking trashed phrases as completed?
     if (completeOnTrash && StringUtils.equals(LABEL_TRASH, label))
     {
       // only if the phrase hasn't been voted on yet
-      if (voteCountForPhrase <= 0)
+      int voteCountForPhrase = votesDao.getVoteCountForPhrase(phraseId);
+      if (voteCountForPhrase <= 1)
```

Comment:
Is the upsert guaranteed to complete before we get the vote counts?

Reply:
it blocks